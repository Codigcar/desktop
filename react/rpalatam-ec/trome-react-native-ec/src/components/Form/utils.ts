import { useCallback, useState } from 'react'

import { FormErrorsProps, FormErrorsReturn, ValidationErros } from './types'

export const useFormErrors = ({
  ref: formRef,
  schema,
}: FormErrorsProps): FormErrorsReturn => {
  const [validationErrors, setValidationErros] = useState<ValidationErros>({})

  const handleChangeField = useCallback(
    async (name, value) => {
      const error = formRef.current?.getFieldError(name)
      const hasError = error && error.length > 0
      try {
        await schema.validate({ [name]: value })
        if (hasError) {
          formRef.current?.setFieldError(name, '')
          setValidationErros((prev) => ({ ...prev, [name]: '' }))
        }
      } catch (err) {
        if (err.inner[name] && (!error || error !== err.inner[name])) {
          formRef.current?.setFieldError(name, err.inner[name])
          setValidationErros((prev) => ({ ...prev, [name]: err.inner[name] }))
        } else if (!err.inner[name] && hasError) {
          formRef.current?.setFieldError(name, '')
          setValidationErros((prev) => ({ ...prev, [name]: '' }))
        }
      }
    },
    [formRef, schema],
  )

  const setErrors = useCallback(
    (errors) => {
      formRef.current?.setErrors(errors)
      setValidationErros(errors)
    },
    [formRef],
  )

  return {
    errors: validationErrors,
    setFieldError: handleChangeField,
    setErrors,
  }
}
