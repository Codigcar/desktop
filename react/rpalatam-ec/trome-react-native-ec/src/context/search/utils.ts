import { FilterKeys } from './search'
import { Story } from '../../entities/Story'

export type SearchState = {
  filters: Map<FilterKeys, string>
  hasMore: boolean
  hasError: boolean
  page: number
  status: 'idle' | 'started' | 'done'
  results: Story[]
  orderInverse: boolean
}

export const initialState: SearchState = {
  filters: new Map(),
  hasMore: true,
  hasError: false,
  page: 0,
  status: 'idle',
  results: [],
  orderInverse: false,
}

type SearchAction =
  | { type: 'init' }
  | { type: 'failure' }
  | { type: 'reset' }
  | { type: 'restore' }
  | { type: 'clear'; payload?: { searchTerm?: string } }
  | {
      type: 'success'
      payload: {
        results: Story[]
        filters: Map<FilterKeys, string>
        orderInverse: boolean
        hasMore: boolean
        page: number
      }
    }

export const searchReducer = (
  state = initialState,
  action: SearchAction,
): SearchState => {
  switch (action.type) {
    case 'init':
      return { ...state, hasError: false, status: 'started' }
    case 'success': {
      const {
        payload: { results, filters, orderInverse, hasMore, page },
      } = action
      return {
        ...state,
        page,
        orderInverse,
        results,
        filters,
        hasMore,
        status: 'done',
        hasError: false,
      }
    }
    case 'clear': {
      const { payload } = action
      const nextFilters = new Map<FilterKeys, string>()
      if (payload?.searchTerm !== undefined) {
        nextFilters.set('term', payload?.searchTerm)
      }
      return { ...initialState, filters: nextFilters }
    }
    case 'failure':
      return { ...state, hasError: true, status: 'done' }
    case 'reset': {
      const hasResults = state.results.length > 0
      return { ...state, status: hasResults ? 'done' : 'idle' }
    }
    case 'restore': {
      return initialState
    }
    default:
      return state
  }
}
