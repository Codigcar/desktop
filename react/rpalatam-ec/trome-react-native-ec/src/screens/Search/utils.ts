export const computeEstimatedDate = (date: number): string => {
  const currentDate = new Date()
  const approx = currentDate.getDate() + date
  currentDate.setDate(approx)

  const month = `0${currentDate.getUTCMonth() + 1}`.slice(-2)
  const day = `0${currentDate.getUTCDate()}`.slice(-2)
  const year = currentDate.getUTCFullYear()

  return `${year}-${month}-${day}`
}

export const parseDate = (date?: Date): string | null => {
  if (!date) return null
  const month = `0${date.getUTCMonth() + 1}`.slice(-2)
  const day = `0${date.getUTCDate()}`.slice(-2)
  const year = date.getUTCFullYear()

  return `${year}-${month}-${day}`
}

export const timeFrames = [
  {
    value: '',
    label: 'Cualquier fecha',
  },
  {
    value: '-1',
    label: 'Últimas 24 horas',
  },
  {
    value: '-7',
    label: 'Última semana',
  },
  {
    value: '-30',
    label: 'Último mes',
  },
  {
    value: 'custom',
    label: 'Personalizado',
  },
]
