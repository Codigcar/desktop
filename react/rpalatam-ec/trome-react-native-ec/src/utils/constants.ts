import NativeConfig from 'react-native-config'

import { App } from './config'

const IS_PRODUCTION = NativeConfig.APP_ENVIRONMENT === 'production'

export const STORE_FAVORITES = '@storage_favorites'
export const STORE_HOME_SCREEN_SHOWN = '@app_home_screen_shown'
export const STORE_SCHEME = '@app_scheme'
export const STORE_CATEGORIES = '@user_categories'
export const STORAGE_OPEN_INTERSTITIAL = '@storage_open_interstitial'
export const STORE_INTERESTS = '@storage_interests'
export const STORE_SEARCH_HISTORY = '@storage_search_history'
export const STORE_TOPICS = '@storage_topics'
export const SUBSCRIPTION_ONBOARDING_DATE = '@subscription_onboarding_date'
export const SIGNUP_ONBOARDING_SHOWN = '@signup_onboarding_shown'
export const FROM_DATE_SUBSCRIPTION_ONBOARDING = new Date(2022, 9, 1).getTime() // 1 octubre de 2022

export type CategoryFilter = {
  whitelist: Record<string, string>
  blacklist: Record<string, string>
}

const PORTRAIT_CATEGORY = {
  ad: 'principal',
  key: 'portada',
  label: 'Portada',
  path: '/category/portada',
  type: 'category',
}

const LAST_NEWS_CATEGORY = {
  ad: 'default',
  key: 'ultimo',
  label: 'Lo Ãšltimo',
  path: '/category/ultimo',
  type: 'category',
}

export const DEFAULT_CATEGORIES = App.select({
  gestion: [],
  default: [PORTRAIT_CATEGORY, LAST_NEWS_CATEGORY],
})

export const DEFAULT_CATEGORY_FILTER = App.select({
  elcomercio: {
    blacklist: {
      'link-M8VGH671510M59XRHTM9RCEBHR': 'ultimas-noticias',
      'link-EQ855NA75531QEP65VRV0RGV1M': 'portada',
      'link-UNUC2AP10H0HFA9V3HM4YAZE84': 'blogs',
      'link-Z2DZAKBU3D2XZ1YCUVNHNYAZ1M': 'publirreportaje',
      'link-EAD20B60MD64D5CKE0M5NNU2KR': 'colecciones-el-comercio',
      'link-9YQN2G2D6T7YDDCN8ZQCD4TM7M': 'las-mas-leidas',
      'link-E1W59X02NT3XK1RW7XTR9URCCR': 'tecnologia-y-ciencias',
      'link-UWWK4AQ5KT4N38M36QU6NKDTXM': 'edicion-impresa',
    },
    whitelist: {
      'link-H2UWFWUQ712HB883Q5DEXFEC6C': '/tecnologia-e-sports',
    },
  },
  gestion: {
    blacklist: {
      '/gestion-tv': 'gestion-tv',
      'link-8GR1QGUUET4YD0XHVUEND11FE8': 'portada',
    },
    whitelist: {
      'link-P5WJUUAHN956Z4WZ9486W4YTUC': '/plusg',
    },
  },
  peru21: {
    blacklist: {
      '/peru21tv': 'peru21-tv',
      'link-TKYHZKC0DX3KDA2PEXVBR9KFJC': 'ultimas-noticias',
    },
    whitelist: {},
  },
  trome: {
    blacklist: {
      '/dr-trome': 'dr-trome',
      '/edicion-impresa': 'edicion-impresa',
      '/fotos': 'fotos',
      '/malcriadas': 'malcriadas',
    },
    whitelist: {},
  },
  depor: {
    blacklist: {
      '/archivo': 'archivo',
    },
    whitelist: {},
  },
})

export const AD_NEWS_LIST = {
  free: {
    positions: [1, 7, 13, 19, 25, 31, 37, 43],
    size: 'mediumRectangle',
    sizes: ['mediumRectangle', 'largeBanner', 'banner'],
  },
  premium: {
    positions: [],
    size: 'banner',
    sizes: ['banner'],
  },
}

export const ORIGIN = App.select({
  depor: IS_PRODUCTION
    ? 'https://depor.com'
    : 'https://elcomercio-depor-sandbox.cdn.arcpublishing.com',
  elcomercio: IS_PRODUCTION
    ? 'https://elcomercio.pe'
    : 'https://elcomercio-elcomercio-sandbox.cdn.arcpublishing.com',
  gestion: IS_PRODUCTION
    ? 'https://gestion.pe'
    : 'https://elcomercio-gestion-sandbox.cdn.arcpublishing.com',
  peru21: IS_PRODUCTION
    ? 'https://peru21.pe'
    : 'https://elcomercio-peru21-sandbox.cdn.arcpublishing.com',
  trome: IS_PRODUCTION
    ? 'https://trome.pe'
    : 'https://elcomercio-trome-sandbox.cdn.arcpublishing.com',
})

export const LEGAL_URLS = App.select({
  depor: {
    data: 'https://depor.com/tratamiento-de-datos/',
    policies: 'https://depor.com/politicas-privacidad/',
    terms: 'https://depor.com/terminos-servicio/',
  },
  elcomercio: {
    data: 'https://elcomercio.pe/tratamiento-de-datos/',
    policies: 'https://elcomercio.pe/politicas-privacidad/',
    terms: 'https://elcomercio.pe/terminos-y-condiciones/',
  },
  gestion: {
    data: 'https://gestion.pe/tratamiento-de-datos/',
    policies: 'https://gestion.pe/politica-de-privacidad/',
    terms: 'https://gestion.pe/terminos-y-condiciones/',
  },
  peru21: {
    data: 'https://peru21.pe/tratamiento-de-datos/',
    policies: 'https://peru21.pe/politicas-de-privacidad/',
    terms: 'https://peru21.pe/terminos-y-condiciones/',
  },
  trome: {
    data: 'https://trome.pe/tratamiento-de-datos/',
    policies: 'https://trome.pe/politica-de-privacidad/',
    terms: 'https://trome.pe/terminos-y-condiciones/',
  },
})

export const SUBSCRIPTION_LANDING_URL = `${ORIGIN}/suscripciones`
