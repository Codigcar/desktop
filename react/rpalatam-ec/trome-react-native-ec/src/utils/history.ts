import { HomeWebviewRef } from './refs'
import type { Hash, Path, Pathname, Search } from 'history'

/**
 * The pieces of a URL path.
 */
export interface PathPieces {
  pathname?: Pathname
  search?: Search
  hash?: Hash
}

/**
 * A URL path or an object that contains the pieces of a URL path.
 */
export type To = Path | PathPieces

/**
 * An object that is used to associate some arbitrary data with a location, but
 * that does not appear in the URL path.
 *
 * @see https://developer.mozilla.org/en-US/docs/Web/API/History/state
 */
export type State = Record<string, unknown> | null

/**
 * Methods of a history object exposed by react-router
 *
 * @see https://v5.reactrouter.com/web/api/history
 */
interface History {
  /**
   * Pushes a new location onto the history stack, increasing its length by one.
   * If there were any entries in the stack after the current one, they are
   * lost.
   *
   * @param to - The new URL
   * @param state - Data to associate with the new location
   */
  push(to: To, state?: State): void

  /**
   * Replaces the current location in the history stack with a new one.  The
   * location that was replaced will no longer be available.
   *
   * @param to - The new URL
   * @param state - Data to associate with the new location
   */
  replace(to: To, state?: State): void

  /**
   * Navigates to the previous entry in the stack. Identical to go(-1).
   *
   * Warning: if the current location is the first location in the stack, this
   * will unload the current document.
   */
  back(): void
}

/**
 * Methods of a history object exposed in home webview to modify
 * web location from native side
 */
const history: History = {
  push(...args) {
    const str = args.map((arg) => JSON.stringify(arg)).join(',')
    const js = `
      if (window.appHistory) window.appHistory.push(${str});
    `
    HomeWebviewRef.current?.injectJavaScript(`(function(){${js}})();`)
  },
  replace(...args) {
    const str = args.map((arg) => JSON.stringify(arg)).join(',')
    const js = `
      if (window.appHistory) window.appHistory.replace(${str});
    `
    HomeWebviewRef.current?.injectJavaScript(`(function(){${js}})();`)
  },
  back() {
    const js = `
     window.appHistory?.goBack();
    `
    HomeWebviewRef.current?.injectJavaScript(`(function(){${js}})();`)
  },
}

export default history
