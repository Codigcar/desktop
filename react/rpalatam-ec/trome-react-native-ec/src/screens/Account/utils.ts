import { App } from '../../utils/config'
import * as Schema from '../../utils/validation'

const name = Schema.string()
  .matches(/\d+/i, 'Este campo no puede contener números')
  .matches(/\D{51,}/i, 'Máximo 50 caracteres')

export const ProfileSchema = Schema.object({
  first_name: App.select({
    elcomercio: name.clone().min(2).required(),
    default: name.clone().required(),
  }),
  last_name: name.clone().min(2).required(),
  second_last_name: name.clone().min(2),
  email: Schema.string().required().email(),
  mobile_phone: App.select({
    elcomercio: Schema.string()
      .matches(/\D+/i, 'Se aceptan solo números')
      .min(6, 'Mínimo 6 dígitos')
      .max(12, 'Máximo 12 dígitos'),
    default: Schema.string()
      .min(9, 'Mínimo 9 dígitos')
      .max(13, 'Máximo 13 dígitos'),
  }),
  user_metadata: {
    documentNumber: App.select({
      elcomercio: Schema.string().required(),
      default: Schema.string(),
    })
      .matches(/\s+/i, 'No se permiten espacios en blanco')
      .when('user_metadata.documentType', (value, schema) => {
        if (value === undefined) return schema
        if (value === 'DNI') {
          return schema
            .required()
            .matches(/\D+/i, 'Se aceptan solo números')
            .min(8, 'Se aceptan 8 dígitos')
            .max(8, 'Se aceptan 8 dígitos')
        }
        return schema
          .required()
          .min(9, 'Mínimo 9 dígitos')
          .max(20, 'Máximo 20 dígitos')
      }),
    documentType: Schema.string(),
  },
})
const gendersBase = [
  { label: 'Hombre', value: 'MALE' },
  { label: 'Mujer', value: 'FEMALE' },
]
export const OPTIONS = {
  country: [{ label: 'Perú', value: '260000' }],
  civilStatus: [
    { label: 'Soltero (a)', value: 'SO' },
    { label: 'Casado (a)', value: 'CA' },
    { label: 'Divorciado (a)', value: 'DI' },
    { label: 'Viudo (a)', value: 'VI' },
  ],
  document: [
    { label: 'DNI', value: 'DNI' },
    { label: 'CEX', value: 'CEX' },
    { label: 'CDI', value: 'CDI' },
  ],
  gender: App.select({
    elcomercio: [
      ...gendersBase,
      { label: 'Prefiero no decirlo', value: 'UNKNOWN' },
      { label: 'Otro', value: 'OTHER' },
    ],
    default: gendersBase,
  }),
}
