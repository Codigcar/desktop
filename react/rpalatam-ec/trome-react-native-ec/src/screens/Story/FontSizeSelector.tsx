import Slider from '@react-native-community/slider'
import { useTheme } from '@shopify/restyle'
import React, { useRef } from 'react'
import { Platform } from 'react-native'

import Box from '../../components/box'
import Typography from '../../components/typography'
import { storage } from '../../utils/storage'
import type { Theme } from '../../theme'

const { View } = Box
const { Paragraph } = Typography

type Props = {
  onSizeChange(size: string, prevSize: string): void
}

export enum FontSize {
  xs = 1,
  s = 2,
  m = 3,
  l = 4,
  xl = 5,
}

// Storage key for font size
const key = 'config.fontSize'

export function getFontSize(): keyof typeof FontSize {
  const value = storage.getNumber(key) || FontSize.m
  return FontSize[value] as keyof typeof FontSize
}

const FontSizeSelector: React.FC<Props> = ({ onSizeChange }) => {
  const { colors } = useTheme<Theme>()
  const fontSizeName = useRef(getFontSize())
  const fontSize = FontSize[fontSizeName.current]

  const handleValueChange = (value: number) => {
    onSizeChange(FontSize[value], fontSizeName.current)
    storage.set(key, value)
    fontSizeName.current = FontSize[value] as keyof typeof FontSize
  }

  return (
    <View
      pb={Platform.OS === 'android' ? '2' : '0.75'}
      pt="0.75"
      px="1.5"
      width="100%">
      <View mb="1">
        <Paragraph color="coolGray-700" fontWeight="bold">
          Seleccione el tama√±o de letra
        </Paragraph>
      </View>
      <View alignItems="center" flexDirection="row">
        <Paragraph color="coolGray-700" fontSize="xs" fontWeight="bold">
          Aa
        </Paragraph>
        <View flex={1} px="0.5">
          <Slider
            minimumValue={1}
            maximumValue={5}
            minimumTrackTintColor={colors.link}
            step={1}
            value={fontSize}
            onValueChange={handleValueChange}
            thumbTintColor={Platform.OS === 'android' ? colors.link : 'white'}
            onResponderGrant={() => true}
            hitSlop={{ top: 8, right: 8, bottom: 8, left: 8 }}
          />
        </View>
        <Paragraph color="coolGray-700" fontSize="xl" fontWeight="bold">
          Aa
        </Paragraph>
      </View>
    </View>
  )
}

export default FontSizeSelector
