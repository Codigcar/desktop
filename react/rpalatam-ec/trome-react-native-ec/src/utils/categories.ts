import AsyncStorage from '@react-native-async-storage/async-storage'

import { STORE_CATEGORIES } from './constants'
import type { NavigationItem } from './navigation'

export type NavCategory = NavigationItem & {
  active?: boolean
  required?: boolean
}

/* utils */
const _required = ({ required }: NavCategory) => !!required
const _notRequired = ({ required }: NavCategory) => !required

export const loadLocalCategories = async (): Promise<
  NavCategory[] | undefined
> => {
  const localStr = await AsyncStorage.getItem(STORE_CATEGORIES)
  return localStr ? JSON.parse(localStr) : undefined
}

export const composeCategoriesWithLocal = (
  navigationCategories: NavCategory[],
  localCategories: NavCategory[],
): NavCategory[] => {
  const requiredCategories = navigationCategories.filter(_required)
  const notRequiredCategories = navigationCategories.filter(_notRequired)

  const userCategories = localCategories
    .filter(_notRequired)
    .map(({ path, ...rest }) => ({
      ...rest,
      path: path.includes('/category') ? path : `/category${path}`,
    }))

  const notRequiredCategoriesKeys = notRequiredCategories.map(({ key }) => key)

  const nextSavedCategories: NavCategory[] = []

  userCategories.forEach((userCategory) => {
    if (notRequiredCategoriesKeys.includes(userCategory.key)) {
      const index = notRequiredCategories.findIndex(
        ({ key }) => key === userCategory.key,
      )
      notRequiredCategories.splice(index, 1)
      nextSavedCategories.push(userCategory)
    }
  })

  return [
    ...requiredCategories,
    ...nextSavedCategories,
    ...notRequiredCategories,
  ]
}
