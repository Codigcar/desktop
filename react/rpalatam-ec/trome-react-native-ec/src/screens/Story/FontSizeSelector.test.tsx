import { render } from '@testing-utils/library'
import React from 'react'

import FontSizeSelector, { getFontSize } from './FontSizeSelector'
import { storage } from '../../utils/storage'

describe('font size selector', () => {
  afterEach(() => storage.delete('testBoolean'))

  it('snapshot', () => {
    const { toJSON } = render(<FontSizeSelector onSizeChange={jest.fn} />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('get default font size', () => {
    expect(storage.getNumber('config.fontSize')).toBeUndefined()
    expect(getFontSize()).toBe('m')
  })

  it('get storage font size', () => {
    storage.set('config.fontSize', 1)
    expect(getFontSize()).toBe('xs')
  })
})
