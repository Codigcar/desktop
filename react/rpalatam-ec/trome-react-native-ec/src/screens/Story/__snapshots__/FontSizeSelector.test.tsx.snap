// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`font size selector snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "paddingBottom": 12,
          "paddingHorizontal": 24,
          "paddingTop": 12,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "marginBottom": 16,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2e3443",
              "display": "flex",
              "fontFamily": "NotoSans",
              "fontSize": 16,
              "fontWeight": "700",
              "lineHeight": 24,
            },
            Object {
              "fontFamily": "NotoSans-Bold",
            },
          ]
        }
      >
        Seleccione el tama√±o de letra
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2e3443",
              "display": "flex",
              "fontFamily": "NotoSans",
              "fontSize": 12,
              "fontWeight": "700",
              "lineHeight": 16,
            },
            Object {
              "fontFamily": "NotoSans-Bold",
            },
          ]
        }
      >
        Aa
      </Text>
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flex": 1,
              "paddingHorizontal": 8,
            },
          ]
        }
      >
        <RNCSlider
          disabled={false}
          enabled={true}
          hitSlop={
            Object {
              "bottom": 8,
              "left": 8,
              "right": 8,
              "top": 8,
            }
          }
          inverted={false}
          maximumValue={5}
          minimumTrackTintColor="#0089ff"
          minimumValue={1}
          onChange={[Function]}
          onRNCSliderAccessibilityAction={null}
          onRNCSliderSlidingComplete={null}
          onRNCSliderSlidingStart={null}
          onRNCSliderValueChange={[Function]}
          onResponderGrant={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          step={1}
          style={
            Object {
              "height": 40,
            }
          }
          tapToSeek={false}
          thumbImage={null}
          thumbTintColor="white"
          value={3}
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#2e3443",
              "display": "flex",
              "fontFamily": "NotoSans",
              "fontSize": 20,
              "fontWeight": "700",
              "lineHeight": 28,
            },
            Object {
              "fontFamily": "NotoSans-Bold",
            },
          ]
        }
      >
        Aa
      </Text>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
